drivers/tm1637/tm1637.c

```diff
@@ -0,0 +1,279 @@
+ *
+ * @param[in] dev device descriptor of the display
+ * @param[in] byte byte to transmit
+ */
+static void transmit_byte(const tm1637_t *dev, uint8_t byte)
+{
+    for (int i = 0; i < 8; ++i) {
+        bool value = (byte >> i) & 0x01;
+        gpio_write(dev->params.clk, false);
+        delay();
+        gpio_write(dev->params.dio, value);
+        delay();
+        gpio_write(dev->params.clk, true);
+        delay();
+    }
+
+    /* we do not read the ACK as it is not necessary for normal functionality */
+    gpio_write(dev->params.clk, false);
+    gpio_write(dev->params.dio, true);
+    delay();
+    gpio_write(dev->params.clk, true);
+    delay();
+    gpio_write(dev->params.clk, false);
+    delay();
+}
+
+/**
+ * @brief Transmits the segments array of length 4 to the display
+ *
+ * @param[in] dev device descriptor of the display
+ * @param[in] segments array of length 4 encoding the display's segments
+ */
+static void transmit_segments(const tm1637_t *dev,
+                              const uint8_t *segments,
+                              tm1637_brightness_t brightness)
+{
+    /* transmit the data command first */
+    start(dev);
+    transmit_byte(dev, DATA_COMMAND);
+    stop(dev);
+
+    /**
+     * set the address using the auto increment mode for addresses and
+     * start at zero
+     */
+    start(dev);
+    transmit_byte(dev, ADDRESS_COMMAND);
+
+    /* transmit each bit indiviudally */
+    for (int i = 0; i < DIGIT_COUNT; ++i) {
+        transmit_byte(dev, segments[i]);
+    }
+    stop(dev);
+
+    /* transmit the display state and the brightness */
+    start(dev);
+    transmit_byte(dev, DISPLAY_AND_CONTROL_COMMAND |
+                           brightness | ON_BIT_MASK);
+
+    stop(dev);
+}
+
+/**
+ * @brief Modifies the segments array to enable the colon
+ *
+ * @param[in,out] segments  segments to enable the colon on
+ */
+static void enable_colon(uint8_t *segments)
+{
+    /* the second digit uses the colon */
+    segments[1] |= DOT_BIT_MASK;
+}
+
+int tm1637_init(tm1637_t *dev, const tm1637_params_t *params)
+{
+    assert(params != NULL);
+    assert(dev != NULL);
+
+    /* set the parameters */
+    dev->params = *params;
+
+    if (gpio_init(dev->params.clk, GPIO_OUT)) {
+        return -1;
+    }
+
+    if (gpio_init(dev->params.dio, GPIO_OUT)) {
+        return -1;
+    }
+
+    gpio_write(dev->params.clk, false);
+    gpio_write(dev->params.dio, false);
+
+    return 0;
+}
+
+void tm1637_clear(const tm1637_t *dev)
+{
+    uint8_t segments[DIGIT_COUNT] = { 0, 0, 0, 0 };
+    transmit_segments(dev, segments, TM1637_PW_1_16);
+}
+
+void tm1637_write_number(const tm1637_t *dev, int16_t number,
+                         tm1637_brightness_t brightness, bool colon,
+                         bool leading_zeros)
+{
+    /* with only 4 digits available, this range can't be exceeded */
+    assert(number <= 9999);
+    assert(number >= -999);
+
+    uint8_t segments[DIGIT_COUNT] = { 0, 0, 0, 0 };
```

Comment:
```suggestion
    uint8_t segments[DIGIT_COUNT] = { 0 };
```

does the same, and doesn't hard-code DIGIT_COUNT ;-)